rrset.go

```diff
@@ -1,40 +1,195 @@
 package udnssdk
 
 import (
+	"encoding/json"
 	"fmt"
+	"log"
+	"strings"
+	"time"
 )
 
-// ZonesService handles communication with the Zone related blah blah
+// RRSetsService provides access to RRSet resources
 type RRSetsService struct {
 	client *Client
 }
 
+// Here is the big 'Profile' mess that should get refactored to a more managable place
+
+//type stringProfile StringProfile
+type Metaprofile struct {
+	Context string `json:"@context"`
+}
+
+// UnmarshalJSON does what it says on the tin
+func (sp *StringProfile) UnmarshalJSON(b []byte) (err error) {
+	sp.Profile = string(b)
+	return nil
+}
+
+// MarshalJSON does what it says on the tin
+func (sp *StringProfile) MarshalJSON() ([]byte, error) {
+	if sp.Profile != "" {
+		return []byte(sp.Profile), nil
+	}
+	return json.Marshal(nil)
+}
+
+// GetType does unknown things
+func (sp *StringProfile) GetType() string {
+	if sp.Profile == "" {
+		return ""
+	}
+	var mp Metaprofile
+	err := json.Unmarshal([]byte(sp.Profile), &mp)
+	if err != nil {
+		log.Printf("Error getting profile type: %+v\n", err)
+		return ""
+	}
+	return mp.Context
+}
+
+// GoString returns the StringProfile's Profile.
+func (sp *StringProfile) GoString() string {
+	return sp.Profile
+}
+
+// String returns the StringProfile's Profile.
+func (sp *StringProfile) String() string {
+	return sp.Profile
+}
+
+// StringProfile wraps a Profile string
+type StringProfile struct {
+	Profile string `json:"profile,omitempty"`
+}
+type RDPoolProfile struct {
```

Comment:
All these wrapper types could use comments saying what they're wrapping. At least what URL path/resource, whether it's for request or response, &c